--- a/thirdparty/include/ddscxx/dds/core/detail/ReferenceImpl.hpp
+++ b/thirdparty/include/ddscxx/dds/core/detail/ReferenceImpl.hpp
@@ -60,7 +60,7 @@
 }

 template <typename DELEGATE>
-dds::core::Reference<DELEGATE>::~Reference<DELEGATE>()
+dds::core::Reference<DELEGATE>::~Reference()
 {
 }

--- a/thirdparty/include/ddscxx/dds/sub/detail/TDataReaderImpl.hpp
+++ b/thirdparty/include/ddscxx/dds/sub/detail/TDataReaderImpl.hpp
@@ -511,7 +511,7 @@
 }

 template <typename T>
-dds::sub::detail::DataReader<T>::~DataReader<T>()
+dds::sub::detail::DataReader<T>::~DataReader()
 {
     if (!this->closed) {
         try {
--- a/thirdparty/include/ddscxx/dds/topic/detail/TTopicImpl.hpp
+++ b/thirdparty/include/ddscxx/dds/topic/detail/TTopicImpl.hpp
@@ -209,7 +209,7 @@


 template <typename T>
-dds::topic::detail::Topic<T>::~Topic<T>()
+dds::topic::detail::Topic<T>::~Topic()
 {
     if (!closed) {
         try {
--- a/thirdparty/include/ddscxx/dds/pub/detail/DataWriterImpl.hpp
+++ b/thirdparty/include/ddscxx/dds/pub/detail/DataWriterImpl.hpp
@@ -400,7 +400,7 @@
 }

 template <typename T>
-dds::pub::detail::DataWriter<T>::~DataWriter<T>()
+dds::pub::detail::DataWriter<T>::~DataWriter()
 {
     if (!this->closed) {
         try {
